src/corelibs/U2View/src/ov_sequence/AnnotatedDNAView.cpp

```diff
@@ -786,6 +786,30 @@ void AnnotatedDNAView::sl_updatePasteAction() {
     pasteAction->setEnabled(isEditModeChecked);
 }
 
+void AnnotatedDNAView::sl_relatedObjectRelationChanged() {
+    GObject *o = qobject_cast<GObject *>(sender());
+    CHECK(o != nullptr, );
+    QList<AnnotationTableObject *> currentAnnotations = getAnnotationObjects(false);
+    QList<GObject *> allObjs;
+    foreach (Document *d, AppContext::getProject()->getDocuments()) {
+        allObjs << d->getObjects();
+    }
+    QList<GObject *> objectsToAdd;
+    foreach (GObject *obj, allObjs) {
+        foreach (const GObjectRelation &rel, obj->findRelatedObjectsByRole(ObjectRole_Sequence)) {
+            if (rel.ref.entityRef == o->getEntityRef() && !currentAnnotations.contains(qobject_cast<AnnotationTableObject*>(obj))) {
```

Comment:
Use GObjectUtils::findObjectsRelatedToObjectByRole instead of checking of all objects.